File: Source/Alternet.UI.Pal/PrinterSettings.cpp
#include "PrinterSettings.h"

namespace Alternet::UI
{
    PrinterSettings::PrinterSettings()
    {
    }
    
    PrinterSettings::~PrinterSettings()
    {
    }

    Duplex PrinterSettings::GetDuplex()
    {
        return _duplex;
    }

    void PrinterSettings::SetDuplex(Duplex value)
    {
        _duplex = value;
    }

    int PrinterSettings::GetFromPage()
    {
        return _fromPage;
    }

    void PrinterSettings::SetFromPage(int value)
    {
        _fromPage = value;
    }

    int PrinterSettings::GetToPage()
    {
        return _toPage;
    }

    void PrinterSettings::SetToPage(int value)
    {
        _toPage = value;
    }

    int PrinterSettings::GetMinimumPage()
    {
        return _minimumPage;
    }

    void PrinterSettings::SetMinimumPage(int value)
    {
        _minimumPage = value;
    }

    int PrinterSettings::GetMaximumPage()
    {
        return _maximumPage;
    }

    void PrinterSettings::SetMaximumPage(int value)
    {
        _maximumPage = value;
    }

    PrintRange PrinterSettings::GetPrintRange()
    {
        return _printRange;
    }

    void PrinterSettings::SetPrintRange(PrintRange value)
    {
        _printRange = value;
    }

    bool PrinterSettings::GetCollate()
    {
        return _collate;
    }

    void PrinterSettings::SetCollate(bool value)
    {
        _collate = value;
    }

    int PrinterSettings::GetCopies()
    {
        return _copies;
    }

    void PrinterSettings::SetCopies(int value)
    {
        _copies = value;
    }

    bool PrinterSettings::GetPrintToFile()
    {
        return _printToFile;
    }

    void PrinterSettings::SetPrintToFile(bool value)
    {
        _printToFile = value;
    }

    optional<string> PrinterSettings::GetPrinterName()
    {
        return _printerName;
    }

    void PrinterSettings::SetPrinterName(optional<string> value)
    {
        _printerName = value;
    }

    bool PrinterSettings::GetIsValid()
    {
        if (_printerName == nullopt)
            return true;

        wxPrintData data;
        data.SetPrinterName(wxStr(_printerName.value()));
        return data.IsOk();
    }

    bool PrinterSettings::GetIsDefaultPrinter()
    {
        return _printerName == nullopt;
    }

    optional<string> PrinterSettings::GetPrintFileName()
    {
        return _printFileName;
    }

    void PrinterSettings::SetPrintFileName(optional<string> value)
    {
        _printFileName = value;
    }
}

File: Source/NativeApi/NativeApi/Api/Drawing/Printing/PrinterSettings.cs
using System;

namespace NativeApi.Api
{
    public class PrinterSettings
    {
        public Duplex Duplex { get => throw new Exception(); set => throw new Exception(); }

        public int FromPage { get => throw new Exception(); set => throw new Exception(); }

        public int ToPage { get => throw new Exception(); set => throw new Exception(); }

        public int MinimumPage { get => throw new Exception(); set => throw new Exception(); }

        public int MaximumPage { get => throw new Exception(); set => throw new Exception(); }

        public PrintRange PrintRange { get => throw new Exception(); set => throw new Exception(); }

        public bool Collate { get => throw new Exception(); set => throw new Exception(); }

        public int Copies { get => throw new Exception(); set => throw new Exception(); }

        public bool PrintToFile { get => throw new Exception(); set => throw new Exception(); }

        public string? PrinterName { get => throw new Exception(); set => throw new Exception(); }

        public bool IsValid { get => throw new Exception(); }

        public bool IsDefaultPrinter { get => throw new Exception(); }

        public string? PrintFileName { get => throw new Exception(); set => throw new Exception(); }
    }
}